pkg/condition/conditions_test.go

```diff
@@ -178,6 +154,92 @@ func TestCondition_IsWindowsIPAMEnabled(t *testing.T) {
 	}
 }
 
+func TestCondition_GetPodDataStoreSyncStatus(t *testing.T) {
+	tests := []struct {
+		name            string
+		testBlockedTime int
+		// buffer time is used to consider other execution time during the process
+		testBufferTime int
+		workers        int
+	}{
+		{
+			name:            "worker doesn't wait long enough",
+			testBlockedTime: 5,
+			testBufferTime:  -1,
+			workers:         1,
+		},
+		{
+			name:            "worker waits with a buffer time",
+			testBlockedTime: 5,
+			testBufferTime:  1,
+			workers:         1,
+		},
+		{
+			name:            "three workers check on the flag",
+			testBlockedTime: 5,
+			testBufferTime:  1,
+			workers:         3,
+		},
+		{
+			name:            "large number workers check on the flag",
+			testBlockedTime: 5,
+			testBufferTime:  1,
+			workers:         30, // we want to use an unusual large workers to test routines conflict
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockK8s := mock_k8s.NewMockK8sWrapper(ctrl)
+			conditions := NewControllerConditions(zap.New(), mockK8s)
+			start := time.Now()
+
+			// one routine is keeping the flag as false for 5s and then updates it to true
+			go func() {
```

Comment:
not sure I follow this function. Isn't this running before any workers are created? I am not following why it sets the sync status to false every second

Reply:
This tests two group of routines which are setting (write) the sync field and are getting (read) the field. In real case, pod controller routines keep checking the cache status and set the sync field to true if cache is ready. At the same time, node controller routines keep asking if the pod cache is ready by Getting the status. 
To simulate the case, the test cases create a routine as writer (pod routine) and routines as reader (node routines) to request the lock on the sync field. By using these tests, we can preliminarily test the possibility of lock issue or racing by safeguarding a processing time window.